telemetry/telemetrybuffer.go

```diff
@@ -171,20 +170,23 @@ func (tb *TelemetryBuffer) Connect() error {
 }
 
 // PushData - PushData running an instance if it isn't already being run elsewhere
-func (tb *TelemetryBuffer) PushData() {
-	for {
+func (tb *TelemetryBuffer) PushData(ctx context.Context) {
+	loop := true
+	for loop {
```

Comment:
just curious to understand if goto is not the correct way.

Reply:
`goto` in general is confusing, it isn't commonly seen in `go` because there tends to be a clearer way to accomplish the same effect. 
here, instead of `goto EXIT`, or using this `loop` variable, we could:
`defer tb.Close()` - this is the only code in this func outside of the for/select, and we want it to execute after we break the `for`, 
so then we `return` out of those select branches which should terminate the loop.